pyquil/api/compiler.py

```diff
@@ -161,3 +166,69 @@ def get_job_fn():
+        :return: The JSON payload, with keys "clifford" and "pauli".
+        """
+        max_qubit = max(max(pauli.get_qubits()), max(clifford.get_qubits()))
+        pauli_string = ["I" for _ in range(max_qubit + 1)]
+        for index, term in pauli.operations_as_set():
+            pauli_string[int(index)] = term
+        pauli_string = "".join(pauli_string)
+        payload = {"clifford": clifford.out(),
+                   "pauli": pauli_string}
+        return payload
+
+    def apply_clifford_to_pauli(self, clifford, pauli_in):
+        """
+        Given a circuit that consists only of elements of the Clifford group, return its action on a PauliTerm.
+
+        In particular, for Clifford C, and Pauli P, this returns the PauliTerm representing PCP^{\dagger}.
+
+        :param Program clifford: A Program that consists only of Clifford operations.
+        :param PauliTerm pauli_in: A PauliTerm to be acted on by clifford via conjugation.
+        :return: A PauliTerm corresponding to pauli_in * clifford * pauli_in^{\dagger}
+        """
+        payload = self._pipo_payload(clifford, pauli_in)
+        phase_factor, paulis = post_json(self.session, self.async_endpoint + "/pipo", payload).json()
+        pauli_out = PauliTerm("I", 0, 1.j ** phase_factor)
+        for i, pauli in enumerate(paulis):
+            pauli_out *= PauliTerm(pauli, i)
+        return pauli_out * pauli_in.coefficient
+
+    def _rb_sequence_payload(self, depth, qubits, gateset):
+        """
+        Prepares a JSON payload for generating a randomized benchmarking sequence - see generate_rb_sequence.
+
+        :param int depth: The number of cliffords per rb sequences to generate.
+        :param int qubits: The number of qubits to perform rb on.
+        :param list gateset: A list of Gate objects that make up the gateset to decompose the Cliffords into.
+        :return: The JSON payload, with keys "depth", "qubits", and "gateset".
+        """
+        payload = {"depth": depth,
+                   "qubits": qubits,
+                   "gateset": [gate.out() for gate in gateset]}
+        return payload
+
+    def generate_rb_sequence(self, depth, qubits, gateset):
+        """
+        Construct a randomized benchmarking experiment on the given qubits, decomposing into gateset.
+
+        The JSON payload that is parsed is a list of lists of indices, or Nones. In the former case, they are the index
+         of the gate in the gateset.
+        :param depth: The number of Clifford gates to include in the randomized benchmarking experiement.
+        :param qubits: The list of qubit indices to use.
+        :param gateset: A list of pyquil gates to decompose the Clifford elements into.
+        """
+        payload = self._rb_sequence_payload(depth, qubits, gateset)
+        response = post_json(self.session, self.async_endpoint + "/rb", payload).json()
+        programs = [Program() for _ in response]
```

Comment:
this is not idiomatic. Contruct a `Program()` in the loop and append it to a list, and then drop the `i` which is confusing to read in the presence of `index`

Reply:
Alright!